zarr/tests/test_storage.py

```diff
@@ -1134,6 +1134,35 @@ def create_store(self, normalize_keys=False, key_separator=".", **kwargs):
             key_separator=key_separator)
 
 
+@pytest.mark.skipif(have_fsspec is False, reason="needs fsspec")
+class TestFSStoreFromFilesystem(StoreTests):
```

Comment:
I suppose we should test these in combination with top level zarr objects.

Reply:
Where do those tests belong?

Reply:
I would say right here. I haven't looked to see what other tests do for full integration/roundtrips, but fsspec is certainly a bit special.

Reply:
There are tests with higher-level zarr `Array` objects in `test_core.py` (e.g. `TestArrayWithFSStore`) and for `Group` in `test_hierarchy.py` (see `TestGroupWithFSStore`). Currently those just create an FSStore using a path string as returned by `tempfile.mkdtemp()`